### DISCUSSION TITLE
How to contribute to the birt web-site?

### DISCUSSION THREAD

**Discussion by hvbtup** [eclipse-birt/birt#893] [Q&A]
It seems I don't understand how the birt-website works.
I thought it was decided to create a wiki.
I commited a change to community.md (my PR from January is still stuck) to fix the link to the issues page.
But the commit seems to have no effect on the website.
In a wiki, additional steps (other than "edit this page") should not be necessary to contribute - at least if I have the necessary privileges,  the idea to use a PR for non-commiters is OK to avoid spam etc.

But if additional steps are necessary, this will definitely keep potential contributers from contributing content.

**Comment by wimjongman** [comment]
@chloetz can you answer this?

**Comment by wimjongman** [comment]
>  thought it was decided to create a wiki.

We are using docusaurus. A static website generator

> I commited a change to community.md (my PR from January is still stuck) to fix the link to the issues page.

Can you add a new commit to the branch. I can't seem to retrigger the workflow

**Reply by hvbtup** [reply]
Added a commit.

**Comment by hvbtup** [comment]
PR run failed: Deploy to Github Pages - Update community.md (f722b75)

  /usr/bin/git push origin gh-pages
  remote: Permission to eclipse/birt-website.git denied to github-actions[bot].
  fatal: unable to access 'https://github.com/eclipse/birt-website.git/': The requested URL returned error: 403
  Error: Action failed with "The process '/usr/bin/git' failed with exit code 128"

**Comment by wimjongman** [comment]
Yes, I have filed: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1040

I think we are fine to S&M because it is not a build issue

**Comment by chloetz** [comment]
As far as I remember somebody from Eclipse solved a similar problem last year. The rights wich are granted to me don’t allow to solve the problem at least as far as can see for the moment.

**Comment by wimjongman** [comment]
We aim for the following process.

1. The user sees a typo on the website or wants to add content
2. The user clicks the "Edit this page" link
3. The user commits the change to a local branch in a fork
4. The user creates a PR
5. A committer validates the PR. Specifically, make sure that no malicious code is injected.
6. If the PR is safe, allow starting the Workflow run
7. The workflow run will make the page live
8. Test the changes on the website
9. If ok, Squash and Merge the PR

If not ok, request for changes. If the changes are not coming, then run the workflow again on the main branch. This will rebuild the website from the main branch.

**Comment by chloetz** [comment]
Hi !

We already discussed the workflow you described as I already tried to implement the workflow on github earlier on. On my local environment I have implemented it.

I don’t have the grants to implement it on github. The present workflow was implemented by somebody from eclipse who has the grants. Please ask the person with the adequate grants.

**Reply by wimjongman** [reply]
I have implemented it.

**Comment by chloetz** [comment]
Thank you.
But I don’t understand why I couldn’t implement the changes which work in my local environment.

Von: Wim Jongman ***@***.***
Gesendet: Donnerstag, 24. März 2022 12:58
An: eclipse/birt
Cc: Loetz, Christophe; Mention
Betreff: Re: [eclipse/birt] How to contribute to the birt web-site? (Discussion #893)


I have implemented it.

**Reply by wimjongman** [reply]
Because your fork was 7 commits behind and 8 commits ahead.

**Please if you reply by e-mail, remove the history and only send your new text**

**Comment by hvbtup** [comment]
@chloetz @wimjongman The hurdles for contributing to the documentation are still (too) high.
Oviously with very few exceptions nobody contributes to it and I think that's the main reason.

I think we need an extra page in the documentation (reachable from https://eclipse.github.io/birt-website/docs/how2contribute) that explains how to contribute to the documentation.

For example, the existing pages contain an "edit this page" button, but how can someone add new pages to the documentation?

**Comment by chloetz** [comment]
Hi!

Thank’s for your mail, analysis and question.

Yes it is true, only very few contribute to the BIRT internet pages. I often asked for contents but never got answers. Meanwhile I gave up.

Concerning editing new pages edit either one of the following files and add a new menu item or link:

Main Menu:  docusaurus.config.js
Footer:          docusaurus.config.js
Sidebar  :      sidebars.js
Home Page: src/pages/index.js

Or add an new file in ./docs or  in ./docs/<dir>/ and add an adequate link in one existing file.

I hope the code examples are self-explaining. If not send me a mail and I will write some more.

Best
Christophe



Von: Henning von Bargen ***@***.***
Gesendet: Mittwoch, 25. Mai 2022 10:05
An: eclipse/birt
Cc: Loetz, Christophe; Mention
Betreff: Re: [eclipse/birt] How to contribute to the birt web-site? (Discussion #893)


@chloetz<https://github.com/chloetz> @wimjongman<https://github.com/wimjongman> The hurdles for contributing to the documentation are still (too) high.
Oviously with very few exceptions nobody contributes to it and I think that's the main reason.

I think we need an extra page in the documentation (reachable from https://eclipse.github.io/birt-website/docs/how2contribute) that explains how to contribute to the documentation.

For example, the existing pages contain an "edit this page" button, but how can someone add new pages to the documentation?

**Comment by hvbtup** [comment]
@chloetz Can you create an extra page in the documentation that explains how to contribute to the documentation step-by-step? That information is at the moment somewhat hidden in the comments of this discussion.

**Comment by chloetz** [comment]
I will do so once back from vacation…☺

Von: Henning von Bargen ***@***.***
Gesendet: Mittwoch, 25. Mai 2022 15:09
An: eclipse/birt
Cc: Loetz, Christophe; Mention
Betreff: Re: [eclipse/birt] How to contribute to the birt web-site? (Discussion #893)


@chloetz<https://github.com/chloetz> Can you create an extra page in the documentation that explains how to contribute to the documentation step-by-step? That information is at the moment somewhat hidden in the comments of this discussion.
